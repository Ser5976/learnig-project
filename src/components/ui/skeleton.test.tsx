import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Skeleton } from './skeleton';

describe('Skeleton Component', () => {
  // 1. Базовый рендеринг
  it('renders without crashing', () => {
    const { container } = render(<Skeleton />);
    const skeleton = container.querySelector('[data-slot="skeleton"]');
    expect(skeleton).toBeInTheDocument();
  });

  // 2. Применение классов по умолчанию
  it('applies default classes', () => {
    const { container } = render(<Skeleton />);
    const skeleton = container.querySelector('[data-slot="skeleton"]');
    expect(skeleton).toHaveClass('bg-accent');
    expect(skeleton).toHaveClass('animate-pulse');
    expect(skeleton).toHaveClass('rounded-md');
  });

  // 3. Передача кастомных классов
  it('merges custom className correctly', () => {
    const { container } = render(<Skeleton className="w-10 h-10" />);
    const skeleton = container.querySelector('[data-slot="skeleton"]');
    expect(skeleton).toHaveClass('w-10');
    expect(skeleton).toHaveClass('h-10');
    expect(skeleton).toHaveClass('bg-accent'); // Стандартные классы остаются
  });

  // 4. Передача дополнительных пропсов
  it('accepts additional props', () => {
    const { container } = render(
      <Skeleton aria-label="Loading" style={{ width: '100px' }} />
    );
    const skeleton = container.querySelector('[data-slot="skeleton"]');
    expect(skeleton).toHaveStyle('width: 100px');
  });

  // 5. Snapshot тест
  it('matches snapshot', () => {
    const { asFragment } = render(<Skeleton />);
    expect(asFragment()).toMatchSnapshot();
  });
});
