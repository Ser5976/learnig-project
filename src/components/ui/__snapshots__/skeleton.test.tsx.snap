// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Skeleton Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="bg-accent animate-pulse rounded-md"
    data-slot="skeleton"
  />
</DocumentFragment>
`;
